src/utils/clm_time_manager.F90

```diff
@@ -1484,6 +1488,89 @@ end subroutine get_rest_date
 
   !=========================================================================================
 
+  integer function get_local_time( londeg, offset )
+
+    !---------------------------------------------------------------------------------
+    ! Get the local time for this longitude
+    !
+    ! uses
+    use clm_varcon, only: degpsec, isecspday
+    ! Arguments
+    real(r8)         , intent(in) :: londeg  ! Longitude in degrees
+    integer, optional, intent(in) :: offset  ! Offset from current time in seconds (either sign)
+
+    ! Local variables
+    integer  :: yr, mon, day    ! year, month, day, unused
+    integer  :: secs            ! seconds into the day
+    real(r8) :: lon             ! positive longitude
+    !---------------------------------------------------------------------------------
+    SHR_ASSERT( londeg >= -180.0_r8, "londeg must be greater than -180" )
+    SHR_ASSERT( londeg <= 360.0_r8,  "londeg must be less than 360" )
+    call  get_curr_date(yr, mon, day, secs, offset )
+    lon = londeg
+    if ( lon < 0.0_r8 ) lon = lon + 360.0_r8
+    get_local_time  = secs + nint((lon/degpsec)/real(dtime,r8))*dtime
```

Comment:
The division by dtime here yields incorrect results. See https://github.com/ESCOMP/ctsm/issues/491#issuecomment-422640470 for details.

Reply:
@billsacks and I went over this with @olyson and we figured out why this was done this way. This is just to make sure the code in SurfaceRadiationMod.F90 would match the closest time-step to local noon. This can be handled appropriately with the is_near_local_noon function. So we can remove the dtime logic that's here. The bottom line is that we want to use the method used in Irrigation for going forward. I still plan to have a bit-for-bit tag, that has two different methods and then we'll remove one with a tag that has answer changes.